### PR TITLE
ci: don't run build & test for docs

### LINKED ISSUE #1159: don't run build & test for docs in CI
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.

Hi, folks. I notice that even a typo fix in the doc will trigger the conformance test, which is not so necessary. Therefore I refactor the workflow:
* move the common checks to a separate workflow
* add path scope for the workflows so they will be triggered on demand.

Here is the output:
https://github.com/muyuan0/gateway/pull/1 (coverage test doesn't pass because I haven't setup the codecov).

If permitted, I would submit a PR with it.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by muyuan0**
Fix #1159
Signed-off-by: muyuan0 <127020730+muyuan0@users.noreply.github.com>

### CODE REVIEW

**Review by arkodg — CHANGES_REQUESTED** [review]
lets keep all the `steps` in this file, its easier to reason about it

File: .github/workflows/build_and_test.yaml [hunk 1]
```diff
@@ -5,36 +5,22 @@ on:
     - "main"
     - "release/v*"
     paths-ignore:
-    - "**/*.png"
+    - 'docs/**'
```
Comment (arkodg) [inline .github/workflows/build_and_test.yaml:8]: as I mentioned earlier, we should always run the workflow on a `push`, and not ignore any paths, 
the `docs` path can be ignored on a `pull_request`

**Comment by arkodg** [comment]
rethinking, we can't ignore the `docs` path for pull requests since we need to run lint/spell checks on it.
I prefer if we left things as is

File: .github/workflows/build_and_test.yaml (hunk 1 above)
Comment (muyuan0) [reply .github/workflows/build_and_test.yaml:8]: Sorry by editing the wrong fields!

**Pull request closed without merging**